src/set.cr

```diff
@@ -203,7 +203,7 @@ struct Set(T)
   # Set{1, 1, 3, 5} & Set{1, 2, 3}               # => Set{1, 3}
   # Set{'a', 'b', 'b', 'z'} & Set{'a', 'b', 'c'} # => Set{'a', 'b'}
   # ```
-  def &(other : Set)
+  def &(other : Set) : Set(T)
```

Comment:
Ditto.

Reply:
Same issue as above, though it is a little bit more complicated as the computed hash also needs to be equal.